pytorch_lightning/loops/epoch/evaluation_epoch_loop.py

```diff
@@ -105,9 +105,13 @@ def advance(  # type: ignore[override]
         void(dl_max_batches)
 
         assert self._dataloader_iter is not None
-        batch, self.batch_progress.is_last_batch = next(self._dataloader_iter)
-        if batch is None:
-            raise StopIteration
+        if not isinstance(data_fetcher, DataLoaderIterDataFetcher):
+            batch_idx = self.batch_idx + 1
+            batch, self.batch_progress.is_last_batch = next(self._dataloader_iter)
+            if batch is None:
+                raise StopIteration
```

Comment:
@carmocca @awaelchli i wasn't sure why this raises a stop iteration in eval but not in training:
https://github.com/PyTorchLightning/pytorch-lightning/blob/f41d1e5e5ebb7040a39d137695e818cada9a9234/pytorch_lightning/loops/epoch/training_epoch_loop.py#L162-L164

Reply:
I believe it is not necessary as the fetchers will take care of it. You can try removing it and see if CI passes.